transaction/builder/src/transaction_builder.rs

```diff
@@ -447,6 +447,56 @@ impl<FPR: FogPubkeyResolver> TransactionBuilder<FPR> {
         amount: Amount,
         recipient: &PublicAddress,
         rng: &mut RNG,
+    ) -> Result<TxOutContext, TxBuilderError> {
+        self.add_output_with_tx_private_key(amount, recipient, None, rng)
+    }
+
+    /// Add a non-change output to the transaction, optionally with a specified
+    /// tx_private_key.
```

Comment:
This "optionally" part seems odd to me. I may be failing to recall but don't think many APIs have "do_x_with_y" and the y is optional. One usually calls that method/function to provide y.

Suggest making a common backend that this and add_output() call into. And this common backend allows an optional key. Suggest this private backend be private for now.

```mermaid
flowchart LR
A[add_output_with_tx_private_key] --> B[common_backend]
C[add_output] --> B
```

Reply:
I suggest, take a look at the mobilecoind code that calls this, because in that case, we have an optional RistrettoPrivate parameter. if I have a function call that can just take my optional parameter, that's very convenient. Otherwise I have to write an if else loop and there is more boilerplate and more room for error.

> but don't think many APIs have "do_x_with_y" and the y is optional

I think there are many examples of this in our API when it would be more convenient for the callers.

From my point of view, the only reason to retain `add_output` at all is backwards compatibility, and making it call `add_output_with_tx_private_key` seems fine. We can shorten the name of the new function if you prefer it.

Reply:
Lmk what you prefer nick, happy to go either way

Reply:
I'm fine keeping it as currently proposed, the argument of of not iffing based on function name, just args sold me on it